internal/pkg/externalplugins/cherrypicker/cherrypicker.go

```diff
@@ -0,0 +1,685 @@
+				logger.WithField("preexisting_cherrypick", pr.HTMLURL).Info("PR already has cherrypick")
+				resp := fmt.Sprintf("Looks like #%d has already been cherry picked in %s.", num, pr.HTMLURL)
+				return s.createComment(logger, org, repo, num, comment, resp)
+			}
+		}
+	}
+
+	// Create the branch for the cherry-pick.
+	if err := r.CheckoutNewBranch(newBranch); err != nil {
+		return fmt.Errorf("failed to checkout %s: %w", newBranch, err)
+	}
+
+	// Title for GitHub issue/PR.
+	title = fmt.Sprintf("%s (#%d)[%s]", title, num, targetBranch)
+
+	// Apply the patch.
+	if err := r.Am(localPath); err != nil {
+		errs := []error{fmt.Errorf("failed to `git am`: %w", err)}
+		logger.WithError(err).Warn("failed to apply PR on top of target branch")
+		resp := fmt.Sprintf("#%d failed to apply on top of branch %q:\n```\n%v\n```.", num, targetBranch, err)
+		if err := s.createComment(logger, org, repo, num, comment, resp); err != nil {
+			errs = append(errs, fmt.Errorf("failed to create comment: %w", err))
+		}
+
+		if opts.IssueOnConflict {
+			resp = fmt.Sprintf("Manual cherrypick required.\n\n%v", resp)
+			if err := s.createIssue(logger, org, repo, title, resp, num, comment, nil, []string{requestor}); err != nil {
+				errs = append(errs, fmt.Errorf("failed to create issue: %w", err))
+			}
+		}
+
+		return utilerrors.NewAggregate(errs)
+	}
+
+	push := r.PushToNamedFork
+	if s.Push != nil {
+		push = s.Push
+	}
+
+	// Push the new branch in the bot's fork.
+	if err := push(forkName, newBranch, true); err != nil {
+		logger.WithError(err).Warn("failed to Push chery-picked changes to GitHub")
+		resp := fmt.Sprintf("failed to Push cherry-picked changes in GitHub: %v.", err)
+		return utilerrors.NewAggregate([]error{err, s.createComment(logger, org, repo, num, comment, resp)})
+	}
+
+	// Open a PR in GitHub.
+	cherryPickBody := createCherrypickBody(num, body)
+	head := fmt.Sprintf("%s:%s", s.BotUser.Login, newBranch)
+	createdNum, err := s.GitHubClient.CreatePullRequest(org, repo, title, cherryPickBody, head, targetBranch, true)
+	if err != nil {
+		logger.WithError(err).Warn("failed to create new pull request")
+		resp := fmt.Sprintf("new pull request could not be created: %v.", err)
+		return utilerrors.NewAggregate([]error{err, s.createComment(logger, org, repo, num, comment, resp)})
+	}
+	*logger = *logger.WithField("new_pull_request_number", createdNum)
+	resp := fmt.Sprintf("new pull request created: #%d.", createdNum)
+	logger.Info("new pull request created")
+	if err := s.createComment(logger, org, repo, num, comment, resp); err != nil {
+		return fmt.Errorf("failed to create comment: %w", err)
+	}
+
+	// Copying original pull request labels.
+	excludeLabelsSet := sets.NewString(opts.ExcludeLabels...)
+	for _, label := range pr.Labels {
+		if !excludeLabelsSet.Has(label.Name) {
+			if err := s.GitHubClient.AddLabel(org, repo, createdNum, label.Name); err != nil {
+				return fmt.Errorf("failed to add label %s: %w", label, err)
```

Comment:
Maybe we should continue to try to add other labels.

Reply:
I think we've done that.

Reply:
Fixed.